File: src/BusinessCardSystem.Domain/BusinessCards/Filters/BusinessCardFilter.cs
using BizCardSystem.Domain.Enums;
using BizCardSystem.Domain.Shared.Filter;

namespace BizCardSystem.Domain.BusinessCards.Filters;

public class BusinessCardFilter : Filter
{
    public string? Name { get; set; }
    public Gender? Gender { get; set; }
    public DateTime? DateofBirth { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
}

File: src/BizCardSystem.Presentation/ClientApp/src/app/modules/filter/BusinessCardFilter.ts
import { Direction } from "../enums/Direction";
import { Gender } from "../enums/Gender";

export class BusinessCardFilter 
{
    name?: string;             
    gender?: Gender;           
    dateOfBirth?: Date;        
    email?: string;
    phone?: string;
    pageIndex: number=1;          
    pageSize: number=12;
    sortColumn?: string;        
    sortDirection: Direction=Direction.Desc;   
  }